### PR TITLE
Add Dockerfile to build da mock server 

### PR DESCRIPTION
**Pull request by jbowen93**
Creates a docker image to build the grpc/mockserv. Additionally adds a gh action that builds and publishes the docker image to ghcr.io/celestiaorg/optimint-mockserv.

### CODE REVIEW

File: da/grpc/mockserv/cmd/go.mod [hunk 1]
```diff
@@ -0,0 +1,56 @@
+module gitub.com/celestiaorg/optimint/mockda
+
+go 1.17
+
+require github.com/celestiaorg/optimint v0.2.0
```
Comment (adlerjohn) [inline da/grpc/mockserv/cmd/go.mod:5]: Hmmmmmmm. Isn't this use a specific version of `optimint`? What if we tag a new release? Then we'd hae to manually increment this _after_ release?

File: da/grpc/mockserv/cmd/go.mod [hunk 2]
```diff
@@ -0,0 +1,56 @@
+module gitub.com/celestiaorg/optimint/mockda
```
Comment (tzdybal) [inline da/grpc/mockserv/cmd/go.mod:1]: Why new module is created for mockserv?

**Comment by jbowen93** [comment]
Fixed all my dumb things like making a new go module. Please re review.

**Review by adlerjohn — COMMENTED** [review]
Why was `go.sum` modified? If it's from `go mod tidy`, can you do that in a separate PR?

**Review by tzdybal — APPROVED** [review]
LGTM

**Comment by tzdybal** [comment]
@jbowen93 anything blocking the merge?

**Pull request merged**